meta-mender-tegra/scripts/mender_tegra_test.py

```diff
@@ -0,0 +1,327 @@
+        self.wait_for_device_removal()
+        self.wait_for_device()
+
+    def nvbootctrl_current_slot(self):
+        '''
+        Returns the current boot slot value
+        '''
+        conn = self.get_connection()
+        result = conn.run("nvbootctrl get-current-slot")
+        return result.stdout.strip()
+
+    def check_partition_mismatch(self):
+        '''
+        Function to check boot and rootfs partition mismatch after reboot.
+        To be used after every reboot.
+        '''
+        conn = self.get_connection()
+        boot_slot = self.nvbootctrl_current_slot()
+        rootfs_part = None
+        if boot_slot == '0':
+            rootfs_part = 'A'
+        elif boot_slot == '1':
+            rootfs_part = 'B'
+
+        if rootfs_part != None:
+            result = conn.run(f'grep -h {rootfs_part} /etc/mender/mender.conf /var/lib/mender/mender.conf | cut -d: -f2 | cut -d, -f1 | tr -d \'" \'')
+            result = conn.run(f'df -h | grep {result.stdout.strip()}')
+            if result.return_code != 0:
+                raise RuntimeError("Boot and Rootfs Partition Mismatch detected")
+        else:
+            raise RuntimeError("Cannot Identify Rootfs Partition Slot")
+
+    def reboot_and_verify_slot(self):
+        '''
+        Verifies if the boot slot has not changed after a reboot.
+        Function to be used when we want to verify if simple reboots lead to boot slot change.
+        '''
+        prev_boot_slot = self.nvbootctrl_current_slot()
+        self.reboot()
+        boot_slot = self.nvbootctrl_current_slot()
+        if boot_slot != prev_boot_slot:
+           raise RuntimeError("Boot slot changed from {} to {} across reboots".format(prev_boot_slot,boot_slot))
+
+    def mender_reboot_and_verify_slot(self):
+        '''
+        Verifies if the boot slot has changed on a reboot after a mender update.
+        Function to be used when we want to verify if reboots after mender updates leads to boot slot change.
+        '''
+        prev_boot_slot = self.nvbootctrl_current_slot()
+        self.reboot()
+        boot_slot = self.nvbootctrl_current_slot()
+        if boot_slot == prev_boot_slot:
+           raise RuntimeError("Boot slot not changed from {} to {} after mender update reboot".format(prev_boot_slot,boot_slot))
+
+    def do_single_mender_update(self, rollback_flag):
+        '''
+        Function to mender update a device, check partition mismatch, boot slot change and mender commit.
+        To use when testing single mender update followed by a reboot.
+        @rollback_flag This parameter expects a bool value. False would replicate successful mender update and boot case.
+                        True would replicate rollback test case. This is achieved by creating the sentinel file which prevents
+                        the boot to be marked successful.
+        Note: do_single_mender_update(True) should be followed by check_rollback() method
```

Comment:
This note is only true if you actually want to rollback, right?  The comment makes it seem as though you need to follow with check_rollback, but that function does more than check whether rollback happens it actually does a rollback I think.